test/python/algorithms/minimum_eigensolvers/test_adapt_vqe.py

```diff
@@ -23,13 +23,15 @@
 from qiskit.opflow import PauliSumOp
 from qiskit.primitives import Estimator
 from qiskit.quantum_info import SparsePauliOp
+from qiskit.utils import algorithm_globals
 
 
 class TestAdaptVQE(QiskitAlgorithmsTestCase):
     """Test of the AdaptVQE minimum eigensolver"""
 
     def setUp(self):
         super().setUp()
+        algorithm_globals.random_seed = 42
```

Comment:
Should you add an `addCleanup()` call here to revert the seed change? Since this is a global this means you've changed the seed for any other tests that run after this class which don't explicitly set a seed. Not necessarily a blocker because anything doesn't set a fixed seed shouldn't care what the rng output is, but it was my first thought after seeing this.

Reply:
This is true, but the ship's sailed elsewhere in the algorithm tests - there's loads of them that do this same line in the `setUp` method with no matching `tearDown`.  If we're worried about it, we can possibly revisit after the release and fix all the algorithms tests with this regard.

It might even be worth just adding this logic to `QiskitAlgorithmsTestCase` or even `QiskitTestCase` and getting rid of it in the subclasses, so it's safe by default.

Reply:
It _should_ be able to work, but these things rarely do with 100% reliability, which is something we need in CI.  We have the same problem with the `quantum_info` module as well, where for some valid inputs there's a teeny chance that some internal BLAS routine gets a bit stuck, and there's nothing we can really do to detect / correct for it.  For tests, anything that uses randomness internally should set a seed, so that if it returns a failure, we can debug it, and we don't have flaky CI coverage.

We should still be able to go back - all the pRNGs in use should expose their `state` somehow or another, so the setup/teardown should be able to save that and restore it afterwards.  It might not be identical to setting the seed through this interface, but it should be achievable without much pain.

Reply:
Its just less than clear to me about restoring state - test order would still matter when using the same RNG. I concur with not wanting flaky unit tests (and buggy code that fails to cater to some random aspect properly).